src/lib/coda_graphql/coda_graphql.ml

```diff
@@ -1583,54 +1584,46 @@ module Mutations = struct
              (Currency.Fee.to_string fee)
              (Currency.Fee.to_string User_command.minimum_fee))
     in
-    let%bind memo =
+    let%map memo =
       Option.value_map memo ~default:(Ok User_command_memo.empty)
         ~f:(fun memo ->
           result_of_exn User_command_memo.create_from_string_exn memo
             ~error:"Invalid `memo` provided." )
     in
-    let%map nonce =
-      match nonce with
-      | Some nonce ->
-          Ok (Account.Nonce.of_uint32 nonce)
-      | None -> (
-        match
-          Coda_commands.get_inferred_nonce_from_transaction_pool_and_ledger
-            coda sender
-        with
-        | `Active (Some nonce) ->
-            Ok nonce
-        | `Active None ->
-            Error
-              "Couldn't infer nonce for transaction from specified `sender` \
-               since `sender` is not in the ledger or sent a transaction in \
-               transaction pool."
-        | `Bootstrapping ->
-            Error "Node is still bootstrapping" )
-    in
-    User_command.Payload.create ~fee ~nonce ~valid_until ~memo ~body
-
-  let send_signed_user_command ~coda ~sender ~payload ~signature =
+    { User_command_input.sender
+    ; fee
+    ; nonce_opt
+    ; valid_until
+    ; memo
+    ; body
+    ; sign_choice }
+
+  let send_signed_user_command ~signature ~coda ~nonce_opt ~sender ~memo ~fee
+      ~valid_until ~body =
     let open Deferred.Result.Let_syntax in
-    let%bind command =
-      User_command.create_with_signature_checked signature sender payload
-      |> Result.of_option ~error:"Invalid signature"
+    let%bind user_command_input =
+      create_user_command_input ~nonce_opt ~sender ~memo ~fee ~valid_until
+        ~body ~sign_choice:(`Signature signature)
       |> Deferred.return
     in
-    send_user_command coda command
+    send_user_command coda user_command_input
 
-  let send_unsigned_user_command ~coda ~sender ~payload =
+  let send_unsigned_user_command ~coda ~nonce_opt ~sender ~memo ~fee
+      ~valid_until ~body =
     let open Deferred.Result.Let_syntax in
-    let%bind command =
-      match%bind Deferred.return @@ find_identity ~public_key:sender coda with
+    let%bind sign_choice =
```

Comment:
I suspect all the `bind` and `map` here can be eliminated.

Reply:
They are for `Deferred.Result.t`s